### ISSUE TITLE
Query parameter containing encoded &

### ISSUE THREAD

**Issue by jmini** [renatoathaydes/rawhttp#30]
The query parameter `foo & bar` is encoded as `foo%20%26%20bar`

Request:
```
GET /test?custom=foo%20%26%20bar
```

Try following java code:

```java
String requestText = "GET /test?custom=foo%20%26%20bar HTTP/1.1\n" +
    "Host: localhost:8090\n" +
    "\n";
RawHttp http = new RawHttp();
RawHttpRequest httpRequest = http.parseRequest(requestText);
try (Socket socket = new Socket("localhost", 8090)) {
    httpRequest.writeTo(socket.getOutputStream());
    RawHttpResponse<?> httpResponse = http.parseResponse(socket.getInputStream());
    String responseText = httpResponse.eagerly()
        .toString();
    socket.close();

    System.out.println(responseText);
}
```

**Expected:**

Server-side, 1 query parameter:

* name `custom`
* value (decoded) `foo & bar`

**Current:**

2 parameters:

* first:
  * name `custom`
  * value `foo `
* second:
  * name ` bar`
  * no value

---

I think that this URI rewrite (line 88) might be the root cause:

https://github.com/renatoathaydes/rawhttp/blob/a456bf9c5a6d0b33b31111f12d99b4b2857fd4a4/rawhttp-core/src/main/java/rawhttp/core/RequestLine.java#L81-L93

Stacktrace
```
Thread [main] (Suspended (breakpoint at line 93 in RequestLine))	
	RequestLine.toString() line: 93	
	RequestLine.writeTo(OutputStream) line: 70	
	RawHttpRequest(HttpMessage).writeTo(OutputStream, int) line: 152	
	RawHttpRequest(HttpMessage).writeTo(OutputStream) line: 141	
	RawHttpMain.main(String[]) line: 28	
```

The `pathURI` on line 93 is wrong: 

Is: `"/test?custom=foo%20&%20bar"`
Expected: `"/test?custom=foo%20%26%20bar"`

---

Feedback is welcomed.

I am happy to contribute a PR for this issue, but I would like to know the direction I should take.

**Comment by jmini** [comment]
I have proposed a PR: https://github.com/renatoathaydes/rawhttp/pull/31

**Comment by renatoathaydes** [comment]
Keeping issue open until I make a new release (will add a few more fixes first... maybe in a couple of days I get it out).

**Comment by renatoathaydes** [comment]
Fix released in 2.4.0
